### DISCUSSION TITLE
Calling a POST api request from automation tab.

### DISCUSSION THREAD

**Discussion by aashray16** [Budibase/budibase#11545] [Help] [answered]
I am trying to make a post request after a webhook is triggered from the automation tab. In add automation step i dont see anything related to api. is it possible to achieve this?

**Comment by andz-bb** [comment, marked as answer]
hey @aashray16, you can fire a POST request from an automation using the External Data Connector automation step. 

First you'll need to set up your request as a query in a REST data source.
<img width="946" alt="image" src="https://github.com/Budibase/budibase/assets/110921612/f6bc58ee-72ab-4718-8905-f40a9e675d65">

Once that's done you'll be able to select it in your External Data Connector automation step.
<img width="639" alt="image" src="https://github.com/Budibase/budibase/assets/110921612/2ddfe746-682b-4b4a-8565-37c7fb6a8367">

Hope this helps!

**Reply by aashray16** [reply]
@andz-bb 

Thankyou!!  

When i receive the trigger output, i want to pass some of that output values in the json body of the POST request. I tried to configure my REST API data source but there i am unable to bind the trigger output. Can you help with the same or provide some articles related to that.

**Reply by andz-bb** [reply]
@aashray16 sure, the first step is to add bindings to your request in the data section. You can then pass values for these variables when you make the request from the Design or Automation section.

![image](https://github.com/Budibase/budibase/assets/110921612/0a5119c1-9564-43da-8cd3-93d1b42f41f5)

Above is the bindings being set up and then below is them being used in the JSON request body:

![image](https://github.com/Budibase/budibase/assets/110921612/3234e527-adf3-4af3-8211-a965c339ba6c)

Now that I have the bindings set up for my request, I can use values from my webhook trigger in the request body of my POST:
![image](https://github.com/Budibase/budibase/assets/110921612/9c63839a-735e-47c8-8492-16b8d778dff3)

Hope this makes sense - you can see more info about this in these pages of our docs:

- [Webhook automation trigger](https://docs.budibase.com/docs/webhook)
- [REST query bindings](https://docs.budibase.com/docs/rest-bindings)

**Reply by aashray16** [reply]
Thankyou @andz-bb !! It helped me a lot😊
